elasticsearch-transport/lib/elasticsearch/transport/transport/base.rb

```diff
@@ -42,7 +42,8 @@ def initialize(arguments={}, &block)
           @logger      = options[:logger]
           @tracer      = options[:tracer]
 
-          @sniffer     = options[:sniffer_class] ? options[:sniffer_class].new(self) : Sniffer.new(self)
+          sniffer_class = options[:sniffer_class] ? options[:sniffer_class] : Sniffer
+          @sniffer     = sniffer_class.new(self, logger)
```

Comment:
No _strong_ feelings :), but usually it's hard to say right away if the variable without a prefix is a block-local variable or accessor, and in the initializer, I prefer using the `@` variants.